chef-config/lib/chef-config/config.rb

```diff
@@ -837,6 +837,28 @@ def self.export_no_proxy(value)
       ENV["NO_PROXY"] = value unless ENV["NO_PROXY"]
     end
 
+    # Given a scheme, host, and port, return the correct proxy URI based on the
+    # set environment variables, unless exluded by no_proxy, in which case nil
+    # is returned
+    def self.proxy_uri(scheme, host, port)
+      proxy_env_var = ENV["#{scheme}_proxy"].to_s.strip
+
+      # Check if the proxy string contains a scheme. If not, add the url's scheme to the
+      # proxy before parsing. The regex /^.*:\/\// matches, for example, http://. Reusing proxy
+      # here since we are really just trying to get the string built correctly.
+      proxy = if !proxy_env_var.empty?
+                if proxy_env_var.match(/^.*:\/\//)
```

Comment:
The [old usage](https://github.com/chef/chef/pull/4348/files#diff-19d41c6437108880128d0df418f9b4dcL154) would pull out `*_proxy_user` and `*_proxy_pass` to add to the URI if they weren't embedded - seems like this should also do that to maintain functionality